Database/Stored Procedures/add_or_update_podium.sql

```diff
@@ -0,0 +1,130 @@
+/*==============================================================*/
+/* DBMS name:		FestiBase									*/
+/* PDM version:		6											*/
+/* Last edited:		20-12-2017									*/
+/* Edited by:		Mariusz Blautzik							*/
+/* Procedure:		Insert + Update PODIUM						*/
+/*==============================================================*/
+
+USE FestiBase
+GO
+
+DROP PROCEDURE IF EXISTS sp_add_or_update_podium
+GO
+CREATE PROCEDURE sp_add_or_update_podium
+   @podium_number		 INT = NULL,
+   @festival_number      INT,
+   @tent_number			 INT = NULL,
+   @name				 VARCHAR(50),
+   @construction_width	 INT,
+   @construction_length	 INT,
+   @construction_height	 INT = NULL,
+   @floor_height		 INT = NULL,
+   @capacity			 INT,
+   @floor_load			 INT,
+   @free_span_width		 INT,
+   @free_span_length	 INT,
+   @free_span_height	 INT = NULL,
+   @environment			 TEXT,
+   @insert				 BIT
+AS
+BEGIN
+	BEGIN TRY
+
+		IF (@insert = 1)
+		BEGIN
+			INSERT INTO PODIUM(festival_number, tent_number, "name", construction_width, construction_length, construction_height, 
+			                   floor_height, capacity, floor_load, free_span_width, free_span_length, free_span_height, environment)
+			VALUES( @festival_number,
+					@tent_number,
+					@name,
+					@construction_width,
+					@construction_length,
+					@construction_height,
+					@floor_height,
+					@capacity,
+					@floor_load,
+					@free_span_width,
+					@free_span_length,
+					@free_span_height,
+					@environment)
+
+		END
+	ELSE 
+		BEGIN
+			IF (@podium_number IS NULL OR @podium_number = 0)
+			BEGIN
+				;THROW 50000, '@podium_number cannot be NULL or ZERO if an UPDATE is to be commenced.', 1
+			END
+			IF NOT EXISTS (SELECT 1 FROM PODIUM WHERE podium_number = @podium_number)
```

Comment:
Dit staat er nu twee keer, eronder ook
Daarnaast IF EXISTS, niet if NOT
Maak hier gelijk even een test bij, kijk eventueel naar een andere procedure, daar is dit ook gedaan